### PR TITLE
workflows: Factorize unit tests

### PR DESCRIPTION
**Pull request by martinpitt**
Get rid of the triplication by using a strategy matrix.

### CODE REVIEW

**Comment by martinpitt** [comment]
I removed the three old names as expected tests, and once that lands, I'll add back the three new names.

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -1,48 +1,13 @@
 name: unit-tests
 on: [push, pull_request]
 jobs:
-  gcc:
+  unit-tests:
```
Comment (marusak) [inline .github/workflows/unit-tests.yml:4]: `unit-tests / unit-tests (:latest)`
/ME has temptation to point out we could do better, but really just temptation :)

**Review by marusak — APPROVED** [review]
Nice, like it much better when we don't need to define it all 3 times :)

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (martinpitt) [reply .github/workflows/unit-tests.yml:4]: Oh right! Meh, why do we have to give a name to everything :-) This is hard, ideas appreciated!

**Pull request merged**